pkg/local_object_storage/shard/gc.go

```diff
@@ -97,9 +97,11 @@ func (gc *gc) listenEvents() {
 
 			v.prevGroup.Wait()
 			v.prevGroup.Add(1)
+			gc.wg.Add(1)
```

Comment:
Seems wrong. Rather it should be some delayed `v.prevGroup.Wait()` since `gc.wg` has exactly two things to care about. Duplicating an item in different work groups doesn't seem to be a good practice.